client/packages/electron/src/electron.ts

```diff
@@ -0,0 +1,100 @@
+import { app, BrowserWindow, ipcMain } from 'electron';
+import startDiscovery from 'multicast-dns';
+import { IPC_MESSAGES } from './shared';
+import ip from 'ip';
+import { FrontEndHost, frontEndHostUrl } from '@openmsupply-client/common/src/hooks/useElectronClient';
+
+const SERVICE_NAME = '_omsupply._tcp.local';
+const SERVICE_TYPE = 'TXT';
+const QUERY_INTERVAL = 1000;
+
+const discovery = startDiscovery();
+
+let connectedServer: FrontEndHost | null = null;
+let isDiscovering = false;
+
+// This allows TypeScript to pick up the magic constant that's auto-generated by Forge's Webpack
+// plugin that tells the Electron app where to look for the Webpack-bundled app code (depending on
+// whether you're running in development or production).
+declare const MAIN_WINDOW_WEBPACK_ENTRY: string;
+declare const MAIN_WINDOW_PRELOAD_WEBPACK_ENTRY: string;
+
+// Handle creating/removing shortcuts on Windows when installing/uninstalling.
+if (require('electron-squirrel-startup')) {
+  // eslint-disable-line global-require
+  app.quit();
+}
+
+const start = (): void => {
+  // Create the browser window.
+  const window = new BrowserWindow({
+    height: 600,
+    width: 800,
+    webPreferences: {
+      preload: MAIN_WINDOW_PRELOAD_WEBPACK_ENTRY,
+    },
+  });
+  // and load the index.html of the app.
+  window.loadURL(MAIN_WINDOW_WEBPACK_ENTRY);
+
+  ipcMain.on(IPC_MESSAGES.START_SERVER_DISCOVERY, () => {
+    isDiscovering = true;
+  });
+
+  ipcMain.on(IPC_MESSAGES.CONNECT_TO_SERVER, (_event, server: FrontEndHost) => {
+    isDiscovering = false;
+    // Can change 'frontEndHostUrl(server)' to http://localhost:3003 if you want client to display front end served by webpack
+    window.loadURL(frontEndHostUrl(server));
+    connectedServer = server;
+  });
+
+  ipcMain.handle(
+    IPC_MESSAGES.CONNECTED_SERVER,
+    async _event => connectedServer
+  );
+
+  discovery.on('response', function ({ answers }) {
+    const answer = answers[0];
+    if (!answer) return;
+    if (answer.name !== SERVICE_NAME) return;
+    if (answer.type !== SERVICE_TYPE) return;
+
+    let server: FrontEndHost = JSON.parse(answer.data.toString());
+    server.isLocal = server.ip === ip.address();
+
+    window.webContents.send(IPC_MESSAGES.SERVER_DISCOVERED, server);
+  });
+
+  setInterval(() => {
+    if (!isDiscovering) return;
+
+    discovery.query({
+      questions: [
+        {
+          name: SERVICE_NAME,
+          type: SERVICE_TYPE,
+        },
+      ],
+    });
+  }, QUERY_INTERVAL);
+};
+
+app.on('ready', start);
+
+app.on('window-all-closed', () => {
```

Comment:
Boilerplate had the app 'running' after on mac os even if windows are closed, but was removed, also it had extra to open window from docked app (see first commit)

Reply:
> Boilerplate had the app 'running' after on mac os even if windows are closed,

That's pretty standard behaviour on Mac, wouldn't recommended deviating from it.

Reply:
I've been working on windows for too long.. For me - it's natural that closing all the windows closes the app.

Reply:
Will create an issue. Would need to remove listeners when closing windows otherwise getting errors

Reply:
https://github.com/openmsupply/open-msupply/issues/58